### DISCUSSION TITLE
Async Queries in Dashboards & Charts

### DISCUSSION THREAD

**Discussion by gempir** [apache/superset#29250] [Q&A / Help]
I've setup my environment with celery workers to run queries in the background. 
Following this guide https://superset.apache.org/docs/configuration/async-queries-celery/ 

I can see this working flawlessly looking at flower and the jobs being run, but only for scheduled stuff like alerts/reports or the SQLLab 

But a dashboard or chart will never make use of this. It either fetches from my redis cache or it loads 60s and then times out with a 504 after ~60s

Is there some secret I'm missing how to properly run big analytical queries in the background that I can present on a dashboard? 
In my mind a opening of such a dashboard should trigger a celery job to be scheduled to fetch that data, but nothing like that is ever scheduled. 

Dashboards don't make use of celery at all. 

I see some ideas about cache warmup here, but it seems this has been removed since it's not really ready. Is there no recommended way to run really long queries currently? https://github.com/apache/superset/pull/20269

**Comment by Tim-Hodge** [comment]
Have you enabled the `"GLOBAL_ASYNC_QUERIES": True` feature flag? Once I switched that on as well dashboards worked async with the celery workers. It's not in the docs as GAQ is still kinda beta and due for rework: https://github.com/apache/superset/issues/9190#issuecomment-2205419056

**Reply by gempir** [reply]
Does that not require the usage of the new websocket system? There is much discussion and change around that system, so I did not want to put work into making that system work in my environment.

**Reply by Tim-Hodge** [reply]
It doesn't need websockets, that's an optional extra and will default to polling if ws is not enabled. It sounds like ws is a lot of overhead and the polling appears to work well enough for my use case.
You will need a value set for `GLOBAL_ASYNC_QUERIES_JWT_SECRET` and some other config under `GLOBAL_ASYNC_QUERIES_*` depending on your setup. If I remember any other config I'll reply here.

**Reply by saurshaz** [reply]
Is it (GAQ for charts/dashboards) stable enough for production environment ?
